src/aggregate/aggregate_plan.h

```diff
@@ -162,6 +162,7 @@ struct AGGPlan {
   PLN_ArrangeStep *arrangement;
   PLN_FirstStep firstStep_s;  // Storage for initial plan
   uint64_t steptypes;         // Mask of step-types contained in plan
+  bool hasKnn;                // An indicator that the plan contains a KNN arrange step.
```

Comment:
I'm not sure I understand why we need this attribute in the plan. Why don't mark the arrange step as "local only" and make it more general? This might be useful for other fields in the future